### PR TITLE
Feature/8 add travis ci support

### PR DESCRIPTION
**Pull request by shanmukhananda**
travis related changes are tested ok. 
refer: https://github.com/shanmukhananda/ORB_SLAM2
@paul-michalik 
create travis-ci.org account, link git repo. modify reame.md(line 4) to show build status from your travis account.

### CODE REVIEW

**Comment by shanmukhananda** [comment]
@paul-michalik as discussed i will try travis build without docker image dependency. I will update the changes in this PR once the test is ok.

**Comment by shanmukhananda** [comment]
docker pull is removed. Currently using docker build to bootstrap. 
I cannot directly use bootstrap_linux.sh, because travis is using Ubuntu 14 images, and for some reason its not updating cmake to 3.7 despite of apt-get update.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+sudo: required
+language: cpp
```
Comment (paul-michalik) [inline .travis.yml]: I guess this isn't required when building via Docker. But, probably it doesn't hurt either

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+services:
+  - docker
+before_install:
+  - chmod -R 777 *
+  - docker build --compress --tag=orb-slam2 --file ./Dockerfile .
+script:
+  - docker run -it --rm -v $(pwd):/root/orb-slam2 orb-slam2 /bin/bash -c /root/orb-slam2/build_linux.sh
```
Comment (paul-michalik) [inline .travis.yml]: Is '-it' necessary here? I guess it's not...

File: CMakeLists.txt [hunk 3]
```diff
@@ -64,11 +64,12 @@ target_sources(${PROJECT_NAME} PRIVATE
     bootstrap_windows.bat
     build_windows.bat
     bootstrap_linux.sh
-    build.sh
     build_linux.sh
     build_ros.sh)
-
-target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_11)
```
Comment (paul-michalik) [inline CMakeLists.txt]: But we need c++11, right? So you should set it manually in the conditional statement below for older gcc or clang versions

File: bootstrap_linux.sh [hunk 4]
```diff
@@ -1,59 +1,112 @@
...
+    apt-get -y autoremove
+    apt-get -y autoclean
+    rm -rf /var/lib/apt/lists/*
+}
+
+install_dependencies
+download_packages
+install_opencv
+install_pangolin
+cleanup
```
Comment (paul-michalik) [inline bootstrap_linux.sh]: Nice refactoring!

File: .travis.yml (hunk 1 above)
Comment (shanmukhananda) [reply .travis.yml]: language removed. sudo required to change the permissions.

File: .travis.yml (hunk 2 above)
Comment (shanmukhananda) [reply .travis.yml]: Done. Tested ok. both -it and --rm is removed.

File: CMakeLists.txt (hunk 3 above)
Comment (shanmukhananda) [reply CMakeLists.txt]: c++11 flag is not required with latest compilers. Tested with gcc6, msvc141.

**Comment by shanmukhananda** [comment]
review comments addressed with minor changes further.

**Comment by shanmukhananda** [comment]
@paul-michalik possible optimizations: Opencv3 is getting compiled on every commit, travis is not caching build artifacts  we can avoid it by using pre-configured docker image shanmukhananda/orb-slam2 which is present in docker registry.

**Comment by paul-michalik** [comment]
Let's merge now we can change this later. I'd prefer to not use Dockerized build anyway for travis, maybe we can get this in next iteration.

**Pull request merged**